tests/integration/preprocessor/_io/test_load.py

```diff
@@ -21,153 +22,175 @@ def _create_sample_cube():
-                for attr in attributes:
-                    self.assertTrue(attr not in coord.attributes)
-
-    def test_callback_fix_lat_units(self):
-        """Test callback for fixing units."""
-        cube = _create_sample_cube()
-        temp_file = self._save_cube(cube)
-
-        cubes = load(temp_file)
-        cube = cubes[0]
-        self.assertEqual(1, len(cubes))
-        self.assertTrue((cube.data == np.array([1, 2])).all())
-        self.assertTrue(
-            (cube.coord("latitude").points == np.array([1, 2])).all()
-        )
-        self.assertEqual(cube.coord("latitude").units, "degrees_north")
-
-    @unittest.mock.patch("iris.load_raw", autospec=True)
-    def test_fail_empty_cubes(self, mock_load_raw):
-        """Test that ValueError is raised when cubes are empty."""
-        mock_load_raw.return_value = CubeList([])
-        msg = "Can not load cubes from myfilename"
-        with self.assertRaises(ValueError, msg=msg):
-            load("myfilename")
-
-    @staticmethod
-    def load_with_warning(*_, **__):
-        """Mock load with a warning."""
-        warnings.warn(
-            "This is a custom expected warning",
-            category=UserWarning,
-            stacklevel=2,
-        )
-        return CubeList([Cube(0)])
-
-    @unittest.mock.patch("iris.load_raw", autospec=True)
-    def test_do_not_ignore_warnings(self, mock_load_raw):
-        """Test do not ignore specific warnings."""
-        mock_load_raw.side_effect = self.load_with_warning
-        ignore_warnings = [{"message": "non-relevant warning"}]
-
-        # Warning is not ignored -> assert warning has been issued
-        with self.assertWarns(UserWarning):
-            cubes = load("myfilename", ignore_warnings=ignore_warnings)
-
-        # Check output
-        self.assertEqual(len(cubes), 1)
-        self.assertEqual(cubes[0].attributes, {"source_file": "myfilename"})
-
-    @unittest.mock.patch("iris.load_raw", autospec=True)
-    def test_ignore_warnings(self, mock_load_raw):
-        """Test ignore specific warnings."""
-        mock_load_raw.side_effect = self.load_with_warning
-        ignore_warnings = [{"message": "This is a custom expected warning"}]
-
-        # Warning is ignored -> assert warning has not been issued
-        with self.assertRaises(AssertionError):
-            with self.assertWarns(UserWarning):
-                cubes = load("myfilename", ignore_warnings=ignore_warnings)
-
-        # Check output
-        self.assertEqual(len(cubes), 1)
-        self.assertEqual(cubes[0].attributes, {"source_file": "myfilename"})
+def test_load(tmp_path):
+    """Test loading multiple files."""
+    cube = _create_sample_cube()
+    temp_file = tmp_path / "cube.nc"
+    iris.save(cube, temp_file)
+
+    cubes = load(temp_file)
+
+    assert len(cubes) == 1
+    cube = cubes[0]
+    assert cube.attributes.globals["source_file"] == str(temp_file)
+    np.testing.assert_equal(cube.data, [1, 2])
```

Comment:
Use `assert_array_equal` from the `tests` module here too?

Reply:
Done in https://github.com/ESMValGroup/ESMValCore/pull/2579/commits/24bc1500089e9962c55d4a852b8fabfc3fc166e4.